helix-term/src/commands/typed.rs

```diff
@@ -2460,3 +2472,58 @@ pub(super) fn command_mode(cx: &mut Context) {
     prompt.recalculate_completion(cx.editor);
     cx.push_layer(Box::new(prompt));
 }
+
+fn expand_args<'a>(editor: &mut Editor, args: &'a str) -> Cow<'a, str> {
+    let reg = Regex::new(r"%(\w+)\s*\{(.*)").unwrap();
+    reg.replace(args, |caps: &regex::Captures| {
+        let remaining = &caps[2];
+        let end = find_first_open_right_braces(remaining);
```

Comment:
Should we auto-complete for users? I think `%sh{echo hello` shouldn't work.

Reply:
yes, above command won't expand and will be treated literately . The auto-complete is a very nice to have feature, how about we adding it after this PR.